packages/flutter_tools/test/integration/test_driver.dart

```diff
@@ -496,6 +506,85 @@ class FlutterRunTestDriver extends FlutterTestDriver {
   }
 }
 
+class FlutterTestTestDriver extends FlutterTestDriver {
+  FlutterTestTestDriver(Directory _projectFolder, {String logPrefix}):
+    super(_projectFolder, logPrefix: logPrefix);
+
+  Future<void> test({
+    String testFile = 'test/test.dart',
+    bool withDebugger = false,
+    bool pauseOnExceptions = false,
+    File pidFile,
+    Future<void> Function() beforeStart,
+  }) async {
+    await _setupProcess(<String>[
+        'test',
+        '--machine',
+        '-d',
+        'flutter-tester'
+    ], script: testFile, withDebugger: withDebugger, pauseOnExceptions: pauseOnExceptions, pidFile: pidFile);
+  }
+
+  @override
+  Future<void> _setupProcess(
+    List<String> args, {
+    String script,
+    bool withDebugger = false,
+    bool pauseOnExceptions = false,
+    File pidFile,
+    Future<void> Function() beforeStart,
+  }) async {
+    await super._setupProcess(
+      args,
+      script: script,
+      withDebugger: withDebugger,
+      pauseOnExceptions: pauseOnExceptions,
+      pidFile: pidFile,
+    );
+
+    // Stash the PID so that we can terminate the VM more reliably than using
+    // _proc.kill() (because _proc is a shell, because `flutter` is a shell
+    // script).
+    final Map<String, dynamic> version = await _waitForJson();
+    _procPid = version['pid'];
+
+    if (withDebugger) {
+      final Map<String, dynamic> startedProcess =
```

Comment:
Nit: same line

Reply:
Done, though it's quite a long line now (I prefer longer lines, but I wasn't sure if we were trying to stick within 80).